bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/EscalationValidator.java

```diff
@@ -0,0 +1,149 @@
+    if (isReferredByThrowEvent(element) || isReferredByEndEvent(element)) {
+      validateEscalationCode(element, validationResultCollector);
+    } else if (isReferredByEventSubProcessStartEvent(element)) {
+      validateIfReferredByEventSubProcessStartEvent(element, validationResultCollector);
+    }
+  }
+
+  private void validateEscalationCode(
+      final Escalation element, final ValidationResultCollector validationResultCollector) {
+    if (element.getEscalationCode() == null || element.getEscalationCode().isEmpty()) {
+      validationResultCollector.addError(0, "EscalationCode must be present and not empty");
+    }
+  }
+
+  private void validateIfReferredByEventSubProcessStartEvent(
+      final Escalation element, final ValidationResultCollector validationResultCollector) {
+    final Collection<StartEvent> startEvents =
+        element.getParentElement().getChildElementsByType(SubProcess.class).stream()
+            .flatMap(p -> p.getChildElementsByType(StartEvent.class).stream())
+            .collect(Collectors.toList());
+    final long numReferredStartEvents =
+        startEvents.stream()
+            .flatMap(i -> i.getEventDefinitions().stream())
+            .filter(
+                e ->
+                    e instanceof EscalationEventDefinition
+                        && ((EscalationEventDefinition) e).getEscalation() == element)
+            .count();
+
+    if (numReferredStartEvents > 1) {
+      validationResultCollector.addError(
+          0, "A Escalation cannot be referred by more than one start event");
+    } else if (numReferredStartEvents == 1) {
+      validateEscalationCode(element, validationResultCollector);
+    }
+  }
+
+  private boolean isReferredByEndEvent(final Escalation element) {
+
+    final Collection<EndEvent> endEvents = getAllElementsByType(element, EndEvent.class);
+
+    return endEvents.stream()
+        .flatMap(i -> i.getEventDefinitions().stream())
+        .anyMatch(
+            e ->
+                e instanceof EscalationEventDefinition
+                    && ((EscalationEventDefinition) e).getEscalation() == element);
+  }
+
+  private boolean isReferredByThrowEvent(final Escalation element) {
+
+    final Collection<ThrowEvent> throwEvents = getAllElementsByType(element, ThrowEvent.class);
+
+    return throwEvents.stream()
+        .flatMap(i -> i.getEventDefinitions().stream())
+        .anyMatch(
+            e ->
+                e instanceof EscalationEventDefinition
+                    && ((EscalationEventDefinition) e).getEscalation() == element);
+  }
+
+  private boolean isReferredByEventSubProcessStartEvent(final Escalation element) {
+    final Collection<StartEvent> startEvents =
+        element.getParentElement().getChildElementsByType(Process.class).stream()
+            .flatMap(p -> p.getChildElementsByType(SubProcess.class).stream())
+            .flatMap(p -> p.getChildElementsByType(StartEvent.class).stream())
```

Comment:
❌ This doesn't work for nested subprocesses. Take this test for example:
```java
  @Test
  public void test() {
    final var definition =
        Bpmn.createExecutableProcess()
            .startEvent()
            .subProcess(
                "sp1",
                sp ->
                    sp.embeddedSubProcess()
                        .startEvent()
                        .subProcess(
                            "sp2",
                            sp2 -> {
                              sp2.embeddedSubProcess()
                                  .eventSubProcess()
                                  .startEvent()
                                  .interrupting(false)
                                  .escalation()
                                  .endEvent();

                              sp2.embeddedSubProcess()
                                  .eventSubProcess()
                                  .startEvent()
                                  .interrupting(false)
                                  .escalation()
                                  .endEvent();

                              sp2.embeddedSubProcess().startEvent().endEvent();
                            })
                        .endEvent())
            .endEvent()
            .done();

    // when
    final Record<DeploymentRecordValue> rejectedDeployment =
        ENGINE.deployment().withXmlResource(definition).expectRejection().deploy();

    // then
    Assertions.assertThat(rejectedDeployment)
        .hasRecordType(RecordType.COMMAND_REJECTION)
        .hasRejectionType(RejectionType.INVALID_ARGUMENT)
        .hasRejectionReason("A Escalation cannot be referred by more than one start event");
  }
```

This will fail (if the broken validation in the `EscalationEventDefinitionValidator` has been fixed) because the start event of these nested event subprocess are not found.